src/main/java/org/terasology/climateConditions/ExtremeClimateConditionSystem.java

```diff
@@ -0,0 +1,198 @@
+import org.terasology.logic.health.RegenComponent;
+import org.terasology.logic.health.event.ActivateRegenEvent;
+import org.terasology.logic.health.event.DeactivateRegenEvent;
+import org.terasology.logic.health.event.DoDamageEvent;
+import org.terasology.logic.location.Location;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.particles.components.generators.VelocityRangeGeneratorComponent;
+import org.terasology.physics.events.ImpulseEvent;
+import org.terasology.physics.events.MovedEvent;
+import org.terasology.registry.In;
+/*
+import org.terasology.thirst.ThirstUtils;
+import org.terasology.thirst.component.ThirstComponent;
+import org.terasology.utilities.random.Random;
+
+import static org.terasology.logic.health.RegenAuthoritySystem.BASE_REGEN;
+import static org.terasology.thirst.ThirstAuthoritySystem.THIRST_DAMAGE_ACTION_ID;
+*/
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class ExtremeClimateConditionSystem extends BaseComponentSystem{
+    public static final String FROSTBITE_DAMAGE_ACTION_ID = "Frostbite Damage";
+
+    @In
+    private EntityManager entityManager;
+    @In
+    private DelayManager delayManager;
+    @In
+    private PrefabManager prefabManager;
+    @In
+    private Time time;
+
+    private int healthDecreaseInterval = 20000;
+    private int healthDecreaseAmount= 10;
+    private float thresholdHeight= (float)60;
+    private float defaultRunFactor;
+    private float defaultSpeedMultiplier;
+    private float defaultJumpSpeed;
+    private float reducedRunFactorMultiplier = (float)0.6;
+    private float reducedSpeedMultiplierMultiplier = (float)0.7;
+    private float reducedJumpSpeedMultiplier = (float)0.8;
+
+
+    /*
+    NOTE: This part is for the desert effects and is incomplete.
+
+    @ReceiveEvent(components = {ThirstComponent.class, CharacterMovementComponent.class})
+    public void EnterDesert(OnPlayerSpawnedEvent event, EntityRef character, ThirstComponent thirst, CharacterMovementComponent movement) {
+
+        /*
+        Biome newBiome= event.getNewBiome();
+        if(newBiome.getName() == "Desert")
+        {
+            thirst.sprintDecayPerSecond*=10;
+            thirst.normalDecayPerSecond*=10;
+        }
+
+        thirst.sprintDecayPerSecond*=10;
+        thirst.normalDecayPerSecond*=10;
+        movement.runFactor = 1;
+        movement.speedMultiplier = (float)0.8;
+        movement.jumpSpeed = (float)8;
+    }
+*/
+
+    //The following part is for the Snow effects.
+    @ReceiveEvent(components = {PlayerCharacterComponent.class, CharacterMovementComponent.class})
+    public void ChangeSnowEffect(MovedEvent event, EntityRef player, LocationComponent location, CharacterMovementComponent movement) {
```

Comment:
Snow effects here was supposed to mean the effects in the snow biome. Since I was originally gonna keep both desert effects and snow effects in the same pr.... This was only an indicator till the time I was doing a sort of a rough code (trying to implement effects without the biome events)